Apps/W1/Email - Guest Outlook REST API/Source/Email-Guest-M365-REST-API/src/GuestOutlookAPIHelper.Codeunit.al

```diff
@@ -0,0 +1,287 @@
+    var
+        AADApplication: Record "AAD Application";
+    begin
+        if not CheckIfAADApplicationExists(AADApplication) then
+            exit(false);
+
+        exit(AADApplication."Permission Granted");
+    end;
+
+    procedure AddAccount(var GuestOutlookAccount: Record "LGS Email Guest Outlook Acc."; EmailConnector: Enum "Email Connector"; AccountEmailAddress: Text[250]; AccountName: Text[250]) AccountAdded: Boolean
+    var
+        EmailOutlookAccount: Record "Email - Outlook Account";
+    begin
+        EmailOutlookAccount.Id := CreateGuid();
+        EmailOutlookAccount."Outlook API Email Connector" := EmailConnector;
+        EmailOutlookAccount."Email Address" := AccountEmailAddress;
+        EmailOutlookAccount.Name := AccountName;
+
+        AccountAdded := EmailOutlookAccount.Insert();
+
+        GuestOutlookAccount.Id := EmailOutlookAccount.Id;
+        GuestOutlookAccount.TransferFields(EmailOutlookAccount);
+    end;
+
+    procedure UpdateEmailAccount(GuestOutlookAccount: Record "LGS Email Guest Outlook Acc.")
+    var
+        EmailOutlookAccount: Record "Email - Outlook Account";
+
+    begin
+        if not EmailOutlookAccount.Get(GuestOutlookAccount.Id) then
+            exit;
+
+        EmailOutlookAccount.Name := GuestOutlookAccount.Name;
+        EmailOutlookAccount."Email Address" := GuestOutlookAccount."Email Address";
+        EmailOutlookAccount.Modify();
+    end;
+
+
+    local procedure CreateAzureAppRegistration()
+    var
+        Setup: Record "LGS Guest Outlook - API Setup";
+        AppInfo: ModuleInfo;
+        ClientDescription: Text[50];
+        ContactInformation: Text[50];
+        AADApplication: Record "AAD Application";
+        GuestOutlookAADAppDescriptionTxt: Label 'Guest Outlook Integration';
+    begin
+        if not Setup.Get() then
+            exit;
+
+        if IsNullGuid(Setup.ClientId) then
+            exit;
+
+        NavApp.GetCurrentModuleInfo(AppInfo);
+        ClientDescription := CopyStr(GuestOutlookAADAppDescriptionTxt, 1, MaxStrLen(ClientDescription));
+        ContactInformation := CopyStr(AppInfo.Publisher, 1, MaxStrLen(ContactInformation));
+
+        CreateAzureApp(AADApplication, GuestOutlookAADAppDescriptionTxt, ContactInformation, true);
+
+        AADApplication."App ID" := AppInfo.PackageId;
+        AADApplication."App Name" := AppInfo.Name;
+        AADApplication.Modify();
+    end;
+
+    local procedure GrantConsent() Success: Boolean;
+    var
+        Setup: Record "LGS Guest Outlook - API Setup";
+        AADApplication: Record "AAD Application";
+        OAuth2: Codeunit OAuth2;
+        GraphMgtGeneralTools: Codeunit "Graph Mgt - General Tools";
+        CommonOAuthAuthorityUrlLbl: Label 'https://login.microsoftonline.com/common/adminconsent', Locked = true;
+        ErrorMsgTxt: Text;
+    begin
+        if not Setup.Get() then
+            exit;
+
+        if IsNullGuid(Setup.ClientId) then
+            exit;
+
+        Commit();
```

Comment:
Is this commit necessary? What changes in this transaction are we expecting here that we should commit?

Reply:
It is necessary, because after this the `procedure RequestClientCredentialsAdminPermissions` executes with the `OAuth2ControlAddIn.RunModal()` . At this point it gives an error that runmodal cannot be executed.